client/api/filesystem/v2alpha1/api.proto

```diff
@@ -0,0 +1,165 @@
+
+    // IsSymlink checks if a given path is a symlink.
+    rpc IsSymlink(IsSymlinkRequest) returns (IsSymlinkResponse) {}
+}
+
+message PathExistsRequest {
+    // The path whose existence we want to check in the host's filesystem
+    string path = 1;
+}
+
+message PathExistsResponse {
+    // Indicates whether the path in PathExistsRequest exists in the host's filesystem
+    bool exists = 1;
+}
+
+message MkdirRequest {
+    // The path to create in the host's filesystem.
+    // All special characters allowed by Windows in path names will be allowed
+    // except for restrictions noted below. For details, please check:
+    // https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+    // Non-existent parent directories in the path will be automatically created.
+    // Directories will be created with Read and Write privileges of the Windows
+    // User account under which csi-proxy is started (typically LocalSystem).
+    //
+    // Restrictions:
+    // Only absolute path (indicated by a drive letter prefix: e.g. "C:\") is accepted.
+    // Depending on the context parameter of this function, the path prefix needs
+    // to match the paths specified either as kubelet-csi-plugins-path
+    // or as kubelet-pod-path parameters of csi-proxy.
+    // The path parameter cannot already exist in the host's filesystem.
+    // UNC paths of the form "\\server\share\path\file" are not allowed.
+    // All directory separators need to be backslash character: "\".
+    // Characters: .. / : | ? * in the path are not allowed.
+    // Maximum path length will be capped to 260 characters.
+    string path = 1;
+}
+
+message MkdirResponse {
+    // Intentionally empty.
+}
+
+message RmdirRequest {
+    // The path to remove in the host's filesystem.
+    // All special characters allowed by Windows in path names will be allowed
+    // except for restrictions noted below. For details, please check:
+    // https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file
+    //
+    // Restrictions:
+    // Only absolute path (indicated by a drive letter prefix: e.g. "C:\") is accepted.
+    // Depending on the context parameter of this function, the path prefix needs
+    // to match the paths specified either as kubelet-csi-plugins-path
+    // or as kubelet-pod-path parameters of csi-proxy.
+    // UNC paths of the form "\\server\share\path\file" are not allowed.
+    // All directory separators need to be backslash character: "\".
+    // Characters: .. / : | ? * in the path are not allowed.
+    // Path cannot be a file of type symlink.
+    // Maximum path length will be capped to 260 characters.
+    string path = 1;
+
+    // Force remove all contents under path (if any).
+    bool force = 2;
+}
+
+message RmdirResponse {
+    // Intentionally empty.
+}
+
+message RmdirContentsRequest {
+    // The path whose contentes willbe removed in the host's filesystem.
```

Comment:
one minor nit: can you add a space between `will` and `be` above please?

Reply:
done